pkg/iter/iterator.go

```diff
@@ -236,11 +250,19 @@ func (i *heapIterator) Close() error {
 	return nil
 }
 
+func (i *heapIterator) Peek() time.Time {
```

Comment:
Nice I think I can use that here https://github.com/grafana/loki/pull/654/files#diff-5cf7837c33ac75c4bf2be956919db224R455

Does that give the next timestamp without moving the iterator ?

Reply:
Yes. heapiterator already keeps entries pre-fetched from all the underlying iterators. Peek() just returns lowest/higest element considering those pre-fetched entries.

Reply:
I was thinking about suggesting you the same while I was reviewing your pr.

Reply:
I'll rebase your work once you've merged it, thanks !